llvm/lib/Target/AIE/AIEEliminateDuplicatePHI.cpp

```diff
@@ -0,0 +1,148 @@
+//===-- AIEEliminateDuplicatePHI.cpp - Identify and remove duplicate PHI --===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+// Identify and remove duplicate PHI nodes with pointer type in a basic block
+// Duplicate PHI nodes are converted in to copy instructions.
```

Comment:
Comment out of date. We don't remove anything or insert copies, but just make redundant phi nodes dead.